airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams.py

```diff
@@ -0,0 +1,204 @@
+        self._config = config
+        super().__init__(*args, **kwargs)
+
+    url_base = "https://advertising-api.amazon.com/"
+    model = None
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """
+        This method should return a Mapping (e.g: dict) containing whatever information required to make paginated requests. This dict is passed
+        to most other methods in this class to help you form headers, request bodies, query params, etc..
+
+        For example, if the API accepts a 'page' parameter to determine which page of the result to return, and a response from the API contains a
+        'page' number, then this method should probably return a dict {'page': response.json()['page'] + 1} to increment the page count by 1.
+        The request_params method should then read the input next_page_token and set the 'page' param to next_page_token['page'].
+
+        :param response: the most recent response from the API
+        :return If there is another page in the result, a mapping (e.g: dict) containing information needed to query the next page in the response.
+                If there are no more pages in the result, return None.
+        """
+        return None
+
+    def request_headers(self, *args, **kvargs) -> MutableMapping[str, Any]:
+        return {"Amazon-Advertising-API-ClientId": self._config.client_id}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an object representing single record in the response
+        """
+        resp = loads(response.text)
+        for r in resp:
+            yield r
+
+    def get_json_schema(self):
+        return self.model.schema()
+
+
+class Profiles(AmazonAdsStream):
+    """
+    This stream corresponds to Amazon Advertising API - Profiles
+    https://advertising.amazon.com/API/docs/en-us/reference/2/profiles#/Profiles
+    """
+
+    primary_key = "profileId"
+    model = Profile
+
+    def path(self, **kvargs) -> str:
+        return "v2/profiles"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        for r in super().parse_response(response, **kwargs):
+            profile_id_obj = self.model.parse_obj(r)
+            self.ctx.profiles.append(profile_id_obj)
+            yield r
+
+
+class ContextStream(AmazonAdsStream):
+    """
+    Stream for getting resources based on context set by previous stream.
+    """
+
+    def read_records(self, *args, **kvargs) -> Iterable[Mapping[str, Any]]:
+        for profile in self.ctx.profiles:
+            if profile.accountInfo.type == Types.VENDOR:
+                self.ctx.current_profile_id = profile.profileId
+                for record in super().read_records(*args, **kvargs):
+                    yield record
+
+    def request_headers(self, *args, **kvargs) -> MutableMapping[str, Any]:
+        hdr = super().request_headers(*args, **kvargs)
```

Comment:
Why `hdr` and not `headers`? Why save on variable name? Please use names with more meaning in them.